### PR TITLE
Fix - Future decided no instance nil check

### PR DESCRIPTION
**Pull request by MatheusFranco99**
Fix https://github.com/bloxapp/ssv-spec/issues/235 by adding a `nil` check.

New test:
- [x] Future decided with no running instance

### CODE REVIEW

**Review by GalRogozinski — APPROVED** [review]

File: ssv/spectest/tests/runner/consensus/future_decided_no_instance.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		ssvMessage := &types.SSVMessage{
+			MsgType: types.SSVConsensusMsgType,
+			MsgID:   *(*types.MessageID)(getID(role)),
+			Data:    byts,
+		}
+
+		return ssvMessage
+	}
+
+	multiSpecTest := &tests.MultiMsgProcessingSpecTest{
```
Comment (y0sher) [inline ssv/spectest/tests/runner/consensus/future_decided_no_instance.go:47]: we don't need this test for proposer?

Comment (MatheusFranco99) [reply ssv/spectest/tests/runner/consensus/future_decided_no_instance.go:47]: The proposer and blinded proposer tests are in the end of the file

File: ssv/runner.go [hunk 2]
```diff
@@ -196,7 +196,7 @@ func (b *BaseRunner) basePartialSigMsgProcessing(
 // didDecideCorrectly returns true if the expected consensus instance decided correctly
 func (b *BaseRunner) didDecideCorrectly(prevDecided bool, decidedMsg *qbft.SignedMessage) (bool, error) {
 	decided := decidedMsg != nil
```
Comment (y0sher) [inline ssv/runner.go:198]: I suggest to write it differently, this is how we write stuff lie this in go and I also think is more readable and easier to reason about
```golang
if decidedMsg == nil {
 return false, nil
}
if b.State.RunningInstance == nil {
 return false, errors.New("decided wrong instance")
}

if decidedMsg.Message.Height != b.State.RunningInstance.GetHeight() {
 return false, errors.New("decided wrong instance")
}
```

Comment (y0sher) [reply ssv/runner.go:198]: let me know what you guys think, I know its not a part of this PR but I couldn't help but see this
@MatheusFranco99 , @GalRogozinski , @moshe-blox . its ok also to decide not to do it here.

**Review by y0sher — APPROVED** [review]
small code style comment, approved in case you don't want to do it now

File: ssv/runner.go (hunk 2 above)
Comment (MatheusFranco99) [reply ssv/runner.go:198]: I 100% agree. @GalRogozinski what do you think?

File: ssv/runner.go [hunk 3]
```diff
@@ -195,15 +195,18 @@ func (b *BaseRunner) basePartialSigMsgProcessing(
...
-	decidedRunningInstance := decided && decidedMsg.Message.Height == b.State.RunningInstance.GetHeight()
-
-	if !decided {
+	if decidedMsg == nil {
 		return false, nil
 	}
-	if !decidedRunningInstance {
+
+	if b.State.RunningInstance == nil {
+		return false, errors.New("decided wrong instance")
```
Comment (olegshmuelov) [inline ssv/runner.go:203]: we should return different err message here

Comment (MatheusFranco99) [reply ssv/runner.go:203]: I agree

**Pull request merged**